pkg/api/store/cluster/cluster_store.go

```diff
@@ -166,9 +167,49 @@ func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
 		values.PutValue(data, m, managementv3.ClusterFieldAnnotations)
 	}
 
+	if err = setInitialConditions(data); err != nil {
+		return nil, err
+	}
+
 	return r.Store.Create(apiContext, schema, data)
 }
 
+func setInitialConditions(data map[string]interface{}) error {
+	if data[managementv3.ClusterStatusFieldConditions] == nil {
+		data[managementv3.ClusterStatusFieldConditions] = []map[string]interface{}{}
+	}
+
+	conditions, ok := data[managementv3.ClusterStatusFieldConditions].([]map[string]interface{})
+	if !ok {
+		return fmt.Errorf("unable to parse field \"%v\" type \"%v\" as \"[]map[string]interface{}\"",
+			managementv3.ClusterStatusFieldConditions, reflect.TypeOf(data[managementv3.ClusterStatusFieldConditions]))
+	}
+
+	if data["amazonElasticContainerServiceConfig"] != nil || data["azureKubernetesServiceConfig"] != nil ||
```

Comment:
Doesnt catch "pluggable" drivers because they have unknown config keys.
Look for a key with *Config instead of explicit known configs